lambdas/service/app.py

```diff
@@ -1757,6 +1751,36 @@ def validate_wait(wait: Optional[str]) -> Optional[int]:
                                                    headers=headers)
 
 
+@app.route('/repository/sources', methods=['GET'], cors=True, method_spec={
+    'summary': 'List available data sources for the selected catalog',
+    'tags': ['Repository'],
+    'parameters': [catalog_param_spec],
+    'responses': {
+        '200': {
+            'description': format_description('''
+                List the sources in the underlying data repository that the
+                current authenticated user is authorized to access.
+            '''),
+            **responses.json_content(
+                schema.object(sources=array_of_object_spec)
```

Comment:
Too generic. This endpoint should return the sources in a format that's consistent with how they are returned by the /index endpoints, and the spec should reflect that specific format.